README.md

```diff
@@ -324,6 +349,18 @@ instead:
 GOFLAGS="-ldflags=-X=main.version=1.2.3" ko publish .
 ```
 
+## How can I set multiple `ldflags`?
+
+Currently, there is a limitation that does not allow to set multiple arguments in `ldflags` using `GOFLAGS`. Using `-ldflags` multiple times also does not work. For this, `ko` supports two ways to have more complex Go build setups:
+
+- Using [`builds` section](#overriding-go-build-settings) in `.ko.yaml`.
+- Using environment variables `KO_GOBUILD_FLAGS` and `KO_GOBUILD_LDFLAGS`.
```

Comment:
Given the config file, do we need this? I think the magic viper stuff allows setting these via automatically generated environment variables:

https://github.com/google/ko/blob/82cabb40bae577ce3bc016e5939fd85889538e8b/pkg/commands/config.go#L133

I guess `KO_GOBUILD_FLAGS` and `KO_GOBUILD_LDFLAGS` would apply globally to all builds, whereas the viper thing would require you to do this _per build_.

Another convenience is that you could specify this on the command line without having to produce an entire config file... even still, I think I'd rather this be a flag rather than an env... WDYT?

I'm not sure if that convenience is worth the introducing the new environment variable or flag, but I'm open to being convinced.

Reply:
To be honest, the environment variables were more like an afterthought and mostly to fix the issue of having `ldflag` arguments that are dynamic, i.e. putting the result of `git describe --tags --always` into a variable. Since it is dynamic,  I figured the configuration file is not exactly the best place.

Anyway, I think that there is the use case for having a way to have dynamic content in your build arguments. The question would then be what is the best way to achieve that. I like that GoReleaser has Go template support for the strings being used, e.g. `"{{ .CommitTimestamp }}"` or `{{ .Version }}`. That would also be an option.

Reply:
>  I like that GoReleaser has Go template support for the strings being used, e.g. "{{ .CommitTimestamp }}" or {{ .Version }}. That would also be an option.

This seems reasonable to me. Where do those values come from? Can we make them work for any go project (not just within git). I know go tooling has some support for various non-git version control systems...

I see a bunch here: https://goreleaser.com/customization/templates/

We can at least start with supporting `.Env`, which should allow folks to hack their way around most problems.

Reply:
Sorry it took so long, but I could not find cycles to work on this in the last week. The template support works for environment variables now. I find that solution much nicer now. Eventually, we can add even more entries to the data map for the template rendering.